utils/bin/lvim

```diff
@@ -1,6 +1,59 @@
-#!/bin/sh
+#!/bin/bash
 
-export LUNARVIM_RUNTIME_DIR="${LUNARVIM_RUNTIME_DIR:-$HOME/.local/share/lunarvim}"
-export LUNARVIM_CONFIG_DIR="${LUNARVIM_CONFIG_DIR:-$HOME/.config/lvim}"
+#
+# GLOBAL
+#
+# 1 is this is a Global system-wide installation, otherwise unset
+#
+if [[ ${BASH_SOURCE[0]} == /usr/* ]]; then
+  GLOBAL=1
+fi
 
-exec nvim -u "$LUNARVIM_RUNTIME_DIR/lvim/init.lua" "$@"
+#
+# LUNARVIM_INSTALL_DIR
+#
+# Where the main lunarvim repo has been cloned to.
+#
+if [ -v GLOBAL ]; then
+  export LUNARVIM_INSTALL_DIR="/usr/local/share/lunarvim/lvim"
+else
+  export LUNARVIM_INSTALL_DIR="$HOME/.local/share/lunarvim/lvim"
+fi
+
+export LUNARVIM_RUNTIME_DIR="$HOME/.local/share/lunarvim/data"
+export LUNARVIM_CACHE_DIR="$HOME/.local/share/lunarvim/cache"
+export LUNARVIM_CONFIG_DIR="$HOME/.config/lvim"
+
+#
+# When runnins as lvim the neovim process MUST never refer to the standard user config:
+# Anything that uses stdpath('data'), stdpath('cache') etc will get:
+#
+# LUNARVIM_x_DIR/nvim
+#
+export XDG_CONFIG_HOME=${LUNARVIM_CONFIG_DIR}
+export XDG_DATA_HOME="${LUNARVIM_RUNTIME_DIR}"
+export XDG_CACHE_HOME="${LUNARVIM_CACHE_DIR}"
```

Comment:
This is also the approach I thought for solving #2944 
Sadly, neovim does not provide any way to configure what those folders are and make that globally available for everyone, it's very frustrating to be honest.
It really matters because it not only relies on the cache, also the plugins installation place and the extensions state/storage is shared across neovim instances, which is a mess.